clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestManagers.java

```diff
@@ -157,6 +157,7 @@ protected RequestManagers create() {
                 CommitRequestManager commit = null;
 
                 if (groupRebalanceConfig != null && groupRebalanceConfig.groupId != null) {
+                    Optional<String> serverAssignor = Optional.ofNullable(config.getString(ConsumerConfig.GROUP_REMOTE_ASSIGNOR_CONFIG));
```

Comment:
If the protocol is not `consumer` but a group remote assignor is set, do we not need to ignore that config as we ignore the auto commit interval when a group ID is set?

See https://github.com/apache/kafka/blob/ee024966d5283ff95935310e4df039069ae7e695/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java#L475

If yes, we also need to test that like here:
https://github.com/apache/kafka/blob/ee024966d5283ff95935310e4df039069ae7e695/clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java#L920
and here:
https://github.com/apache/kafka/blob/ee024966d5283ff95935310e4df039069ae7e695/clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java#L904

Reply:
Kind of. I think we do _not_ want to ignore it if protocol is not consumer. But we need to ignore it in the constructor of `AsyncConsumer`, since the unused config is reported before it is used.